src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/models/ZipFileData.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services.reports.models;
+
+import java.util.List;
+import java.util.Objects;
+
+public abstract class ZipFileData {
+    public final String zipFileName;
+    public final String container;
+    public final List<String> scannableItemDcns;
+    public final List<String> paymentDcns;
+
+    public ZipFileData(
+        String zipFileName,
+        String container,
+        List<String> scannableItemDcns,
+        List<String> paymentDcns
+    ) {
+        this.zipFileName = zipFileName;
+        this.container = container;
+        this.scannableItemDcns = scannableItemDcns;
+        this.paymentDcns = paymentDcns;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof ZipFileData)) {
+            return false;
+        }
+        ZipFileData that = (ZipFileData) o;
+        return zipFileName.equals(that.zipFileName)
+            && container.equals(that.container);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(zipFileName, container);
+    }
+}
```

Comment:
how about, instead of creating this base class and extending it in two places and then using it as a key in a map to find a matching file, we simply do something like this in the service code?

```java
Optional<ReportedZipFile> matchingFile = 
  reportedFiles
    .stream()
    .filter(reported -> reported.zipFileName.equals(actual.zipFileName)
    .filter(reported -> reported.container.equals(actual.container)
    .findFirst();
```

Reply:
Good point. But in this case we will have to iterate across the whole collection of reported files as many times as there are received files (i.e. 1000+ times with what Exela sends us these days)

Reply:
how about just using a `Pair` (storing file name + container) as a key in that map then?